src/Ticket.php

```diff
@@ -4379,7 +4379,7 @@ public function showForm($ID, array $options = [])
                         break;
                 }
                 // Check category / entity validity
-                if (!in_array($cat->fields['entities_id'], getSonsOf('glpi_entities', $options['entities_id']))) {
+                if (!in_array($cat->fields['entities_id'], getSonsAndAncestorsOf('glpi_entities', $options['entities_id']))) {
```

Comment:
What about this scenario:

>Category entity is `Root` with `is_recursive = false`
>Ticket entity is `Root > A`.

With the proposed change, I think the category will be considered as valid despite not being visible from the tickets entity.

Shouldn't the ancestors be checked only if the category is recursive ?